src/query/storages/fuse/src/io/read/bloom/column_filter_reader.rs

```diff
@@ -0,0 +1,147 @@
+    param: LoadParams,
+}
+
+impl BloomColumnFilterReader {
+    pub fn new(
+        index_path: String,
+        column_id: ColumnId,
+        colum_chunk_meta: &ColumnChunkMetaData,
+        operator: Operator,
+    ) -> Self {
+        let meta = colum_chunk_meta.metadata();
+        let cache_key = format!("{index_path}-{column_id}");
+        let loader = Xor8FilterLoader {
+            offset: meta.data_page_offset as u64,
+            len: meta.total_compressed_size as u64,
+            cache_key,
+            operator,
+            column_descriptor: colum_chunk_meta.descriptor().clone(),
+        };
+
+        let cached_reader = CachedReader::new(
+            Xor8Filter::cache(),
+            "bloom_index_data_cache".to_owned(),
+            loader,
+        );
+
+        let param = LoadParams {
+            location: index_path,
+            len_hint: None,
+            ver: 0,
+        };
+
+        BloomColumnFilterReader {
+            cached_reader,
+            param,
+        }
+    }
+
+    pub async fn read(&self) -> Result<Arc<Xor8Filter>> {
+        self.cached_reader.read(&self.param).await
+    }
+}
+
+/// Loader that fetch range of the target object with customized cache key
+pub struct Xor8FilterLoader {
+    pub offset: u64,
+    pub len: u64,
+    pub cache_key: String,
+    pub operator: Operator,
+    pub column_descriptor: ColumnDescriptor,
+}
+
+#[async_trait::async_trait]
+impl Loader<Xor8Filter> for Xor8FilterLoader {
+    async fn load(&self, params: &LoadParams) -> Result<Xor8Filter> {
+        let reader = self.operator.object(&params.location);
+        let bytes = reader
+            .range_read(self.offset..self.offset + self.len)
+            .await?;
+
+        let page_meta_data = PageMetaData {
+            column_start: 0,
+            num_values: 1,
+            compression: Compression::Uncompressed,
+            descriptor: self.column_descriptor.descriptor.clone(),
+        };
+
+        let page_reader = PageReader::new_with_page_meta(
+            std::io::Cursor::new(bytes), /* we can not use &[u8] as Reader here, lifetime not valid */
+            page_meta_data,
+            Arc::new(|_, _| true),
+            vec![],
+            usize::MAX,
+        );
+
+        let decompressor = BasicDecompressor::new(page_reader, vec![]);
+        let column_type = self.column_descriptor.descriptor.primitive_type.clone();
+        let filed_name = self.column_descriptor.path_in_schema[0].to_owned();
+        let field = ArrowField::new(filed_name, DataType::Binary, false);
+        let mut array_iter =
+            column_iter_to_arrays(vec![decompressor], vec![&column_type], field, None, 1)?;
+        if let Some(array) = array_iter.next() {
+            let array = array?;
+            let col =
+                Column::from_arrow(array.as_ref(), &common_expression::types::DataType::String);
+            let bytes = unsafe { col.as_string().unwrap().index_unchecked(0) };
```

Comment:
it's safe if we ensure it has one element.  But it's better to throw exceptions rather than panic.

```
col.as_string().and_then ....
```